r1cs/gadgets/std/src/groups/mod.rs

```diff
@@ -720,6 +759,13 @@ pub(crate) mod test {
         assert_eq!(a_times_gg_fb.get_value().unwrap(), g.mul(&a));
 
         assert!(cs.is_satisfied());
+
+        // safety checks on mul_bits
+        let zero = GG::alloc(cs.ns(|| "Zero"), || Ok(G::zero())).unwrap();
+        assert!(zero.mul_bits(cs.ns(|| "a * 0"),a_bits.iter()).unwrap_err().to_string().contains("Base point is trivial"));
+
+
```

Comment:
Maybe you could add a safety check to not allow multiplication of a non-zero point by the zero scalar.

Reply:
Done, thanks